src/tests/test_collection.py

```diff
@@ -620,6 +620,51 @@ def test_hnsw(client: vecs.Client) -> None:
     assert len(results) == 1
 
 
+def test_index_build_args(client: vecs.Client) -> None:
+    dim = 4
+    bar = client.get_or_create_collection(name="bar", dimension=dim)
+    bar.upsert([("a", [1, 2, 3, 4], {})])
+
+    # Test that default value for nlists is used in absence of index build args
+    bar.create_index(method="ivfflat")  # type: ignore
+    [nlists] = [i for i in bar.index.split("_") if i.startswith("nl")]
+    assert int(nlists.strip("nl")) == 30
+
+    # Test that default value for nlists is used when when incorrect index build
+    # args are supplied
+    bar.create_index(method=IndexMethod.ivfflat, index_arguments=IndexArgsHNSW(), replace=True)  # type: ignore
+    [nlists] = [i for i in bar.index.split("_") if i.startswith("nl")]
+    assert int(nlists.strip("nl")) == 30
+
+    # Test nlists is honored when supplied
+    bar.create_index(method=IndexMethod.ivfflat, index_arguments=IndexArgsIVFFlat(n_lists=123), replace=True)  # type: ignore
+    [nlists] = [i for i in bar.index.split("_") if i.startswith("nl")]
+    assert int(nlists.strip("nl")) == 123
+
+    # Test that default values for m and ef_construction are used in absence of
+    # index build args
+    bar.create_index(method="hnsw", replace=True)  # type: ignore
+    [m] = [i for i in bar.index.split("_") if i.startswith("m")]
+    [ef_construction] = [i for i in bar.index.split("_") if i.startswith("efc")]
+    assert int(m.strip("m")) == 16
+    assert int(ef_construction.strip("efc")) == 64
+
+    # Test that the default values for m and ef_construction are used when
+    # incorrect index build args are supplied
+    bar.create_index(method="hnsw", index_arguments=IndexArgsIVFFlat(n_lists=123), replace=True)  # type: ignore
+    [m] = [i for i in bar.index.split("_") if i.startswith("m")]
+    [ef_construction] = [i for i in bar.index.split("_") if i.startswith("efc")]
+    assert int(m.strip("m")) == 16
+    assert int(ef_construction.strip("efc")) == 64
+
+    # Test m and ef_construnction is honored when supplied
+    bar.create_index(method="hnsw", index_arguments=IndexArgsHNSW(m=12, ef_construction=123), replace=True)  # type: ignore
```

Comment:
could you add cases for each argument being supplied individually pls?
I know there are defaults, just want to cover all the bases